libs/constrained-generators/src/Constrained/Examples/Map.hs

```diff
@@ -66,3 +71,72 @@ rangeSumSize = constrained $ \m ->
       , v ==. 1
       ]
   ]
+
+rngSetSpec :: Specification BaseFn (Map Int Int)
+rngSetSpec = constrained $ \m -> 3 <=. sizeOf_ (rngSet_ m)
+
+richRngSetSpec :: Specification BaseFn (Map Int Int)
+richRngSetSpec =
+  constrained $ \m ->
+    [ assert $ lit (Set.fromList [20, 30]) `subset_` dom_ m
+    , assert $ elem_ (lit 3) (rng_ m)
+    , assert $ elem_ (lit 4) (rng_ m)
+    , assert $ sizeOf_ (dom_ m) >=. 10
+    , assert $ sizeOf_ (dom_ m) <=. 12
+    , assert $ sizeOf_ (rngSet_ m) >=. 4
+    , assert $ sizeOf_ (rngSet_ m) <=. 6
+    , forAll m (\p -> match p (\k v -> [assert $ k >=. v]))
+    ]
+
+rngSetWithHint :: Specification BaseFn (Map Int Int)
+rngSetWithHint = constrained $ \m ->
+  [ assert $ 3 <=. sizeOf_ (rngSet_ m)
+  , genHint 5 m
+  ]
+
+-- Different ways to constrain the size of a Map
+
+w0 :: Specification BaseFn (Map Int Int)
+w0 = constrained $ \m ->
+  [assert $ sizeOf_ m ==. 10]
+
+w1 :: Specification BaseFn (Map Int Int)
+w1 = constrained $ \m ->
+  [assert $ sizeOf_ (dom_ m) ==. 10] -- sizeOf_ works on things that are Sized
+
+w2 :: Specification BaseFn (Map Three Int)
+w2 = constrained $ \m -> size_ (dom_ m) <=. 3 -- size_ works on Sets,
+
+w3 :: Specification BaseFn (Map Int Int)
+w3 = constrained $ \m ->
+  [assert $ sizeOf_ (rng_ m) ==. 10]
+
+w4 :: Specification BaseFn (Map Int Int)
+w4 = constrained $ \m ->
+  [assert $ sizeOf_ (rngSet_ m) ==. 4]
+
+w5 :: Specification BaseFn (Map Int Int)
+w5 = constrained $ \m -> [assert $ m /=. lit mempty]
+
+w6 :: Specification BaseFn (Map Int Int)
+w6 = constrained $ \m -> [assert $ m ==. lit mempty]
```

Comment:
I would like these to have proper names and for all new examples to be added to the test suite if possible.

Reply:
done